app/src/main/java/space/taran/arknavigator/mvp/presenter/ResourcesPresenter.kt

```diff
@@ -116,23 +116,23 @@ class ResourcesPresenter(
 
     fun onViewResume() {
         tagsSelectorPresenter.calculateTagsAndSelection()
-        if (!tagsEnabled) {
-            presenterScope.launch {
-                resetResources(listResources(untaggedOnly = true))
-            }
+        if (this::storage.isInitialized && this::index.isInitialized) {
+            val aggregatedIndex = (index as? AggregatedResourcesIndex)?: return
```

Comment:
1) This is rude break of incapsulation principles.
2) This is not robust logic: when you have an interface you must assume the object to be any of possible implementations. Here, the bug is fixed for sharded index, but not for plain one.
3) Don't touch indexes please, this removal logic doesn't belong to them.

Reply:
I have one thing to add here - why do we have a presenter within a presenter? What is the reason behind this?

Reply:
@darkovski88 good question, maybe @mdrlzy can answer

Reply:
Multiple presenters are needed to separate business logic.
`gridPresenter` is responsible for which resources will be displayed, also handling the click on the resource and sorting.
`tagsSelectorPresenter` is responsible for which tags will be displayed, also handling a click on a tag and calculating a selection.
`ResourcesPresenter` is a screen layer presenter, it is responsible for initializing the index, tags storage, and handling toolbar menu click. It is also needed for communication between `gridPresenter` and `tagsSelectorPresenter`.